language/go/resolve.go

```diff
@@ -228,6 +228,11 @@ func resolveWithIndexGo(ix *resolve.RuleIndex, imp string, from label.Label) (la
 		} else {
 			// Match is ambiguous
 			matchError = fmt.Errorf("multiple rules (%s and %s) may be imported with %q from %s", bestMatch.Label, m.Label, imp, from)
+			matchError = fmt.Errorf(`code in build rule %s contains import %q which matches multiple rules: %s and %s
+Use # gazelle:resolve to disambiguate with one of the following:
+	# gazelle:resolve go %[2]s %[3]s
+	# gazelle:resolve go %[2]s %[4]s
+`, from, imp, bestMatch.Label, m.Label)
```

Comment:
A few nits about wording:

* Just replace "code in build rule" with "rule".
* Replace "contains import" with "imports".
* Replace "Use # gazelle:resolve" with "# gazelle:resolve may be used".

Also, if this is going to be a multi-line error message now, lets report all of the matching rules instead of just two. `bestMatch` can be a list, and we'll append to it in this case. At the end of the loop, if it contains two or more elements, we'll report the error.

Reply:
Replaced, and made the text into a single line error message.
Since I don't really have the time to rewrite the code to be a full-blown recommendation, I put a TODO in for the time being.  A clearer error message and a recommendation what to do is IMHO already an improvement.

Reply:
TBH, as a new user, `Use # gazelle:resolve to disambiguate with one of the following:` *with an indication of what file this text goes into*, would have been easier to immediately leverage.  Your example would have been so much better.

<--- This new user is randomly guessing which `BUILD.bazel` file (at import, or where the targets are defined), or `WORKSPACE`.  Maybe my syntax is wrong.  Maybe spaces matter.  Maybe `@thing//`, or `//`.  The matrix of possibilities would be alleviated with the text of the example above.

Reply:
Hm, once the amount of needed prerequisite knowledge crosses some threshold, it is hard to put all the needed advice into a single error message.  Perhaps adding a URL to a documentation page would work better?  Sadly, few choose to go that route.

Reply:
I get that -- too much hand-holding makes an error too verbose for most users, and I do appreciate the general effort you have to help others, plus specifically this situation you're improving by these PRs.

The use of a URL given right there, I know I would follow it, and it's not as rare as new users (gasp!) reading documentation.  Stick with me a sec: when you're stuck, and this error message is there, and really "gazelle:resolve" doesn't google any working examples, and I cannot find a working example in the `bazelbuild/bazel-gazelle` unit tests that show where to put the config hint, then the only path forward is the URL in the error message.  I mean, as a last resort, it could get used ;)

Reply:
(Note I'm in the same boat as you, as I'm also user of bazel, not a contributor.)  FWIW, I write (other) software for a living, and *anywhere* we introduced URLs to documentation into error messages, or error codes and such, it has always improved user satisfaction.